public/utils/login-utils.tsx

```diff
@@ -26,3 +27,16 @@ export async function validateCurrentPassword(
     password: currentPassword,
   });
 }
+
+export function extractNextUrlFromWindowLocation(): string {
+  const urlParams = new URLSearchParams(window.location.search);
+  let nextUrl = urlParams.get('nextUrl');
+  if (!nextUrl || nextUrl.toLowerCase().includes('//')) {
```

Comment:
in what cases will it have `//`?

Reply:
tbh I'm not exactly sure. I believe it can happen in cases where the serverBasePath is empty.

I yanked the logic from [here](https://github.com/opensearch-project/security-dashboards-plugin/blob/main/public/apps/login/login-page.tsx#L52-L62) which contains a comment:

> // Appending the next url with trailing slash. We do so because in case the serverBasePath is empty, we can simply
    // redirect to '/'.

Reply:
ahh.. got it.